### PR TITLE
fix(aws): EC2 Image Duplicate Records

### PR DESCRIPTION
**Pull request by bbernays**

### CODE REVIEW

File: plugins/source/aws/resources/services/ec2/images.go [hunk 1]
```diff
@@ -33,3 +39,58 @@ func Images() *schema.Table {
...
+}
+
+func resolveImageArn(_ context.Context, meta schema.ClientMeta, resource *schema.Resource, c schema.Column) error {
+	cl := meta.(*client.Client)
+	item := resource.Item.(types.Image)
+	a := arn.ARN{
+		Partition: cl.Partition,
+		Service:   "ec2",
+		Region:    cl.Region,
+		AccountID: aws.ToString(item.OwnerId),
```
Comment (bbernays) [inline plugins/source/aws/resources/services/ec2/images.go:92]: Changed this from 
```go
AccountID: cl.AccountID,
```

File: plugins/source/aws/resources/services/ec2/images.go [hunk 2]
```diff
@@ -33,3 +39,58 @@ func Images() *schema.Table {
...
+		})
+		for pag.HasMorePages() {
+			resp, err := pag.NextPage(ctx)
+			if err != nil {
+				return err
+			}
+			for _, image := range resp.Images {
+				if aws.ToString(image.OwnerId) != c.AccountID {
+					res <- resp.Images
+				}
```
Comment (bbernays) [inline plugins/source/aws/resources/services/ec2/images.go:76]: Added filtering to exclude any Images that are owned by the same accountId

Comment (hermanschaaf) [reply plugins/source/aws/resources/services/ec2/images.go:76]: You mean "to exclude any images that are NOT owned by the same account ID", right?

Comment (hermanschaaf) [reply plugins/source/aws/resources/services/ec2/images.go:76]: Or no... maybe I'm not understanding this properly

Comment (hermanschaaf) [reply plugins/source/aws/resources/services/ec2/images.go:76]: Oh I see, first we list images we own, then images we can see (and filter out the ones we own)

**Review by hermanschaaf — APPROVED** [review]

File: plugins/source/aws/resources/services/ec2/images.go (hunk 2 above)
Comment (bbernays) [reply plugins/source/aws/resources/services/ec2/images.go:76]: Exactly! There is no way to have multiple filters to get both the AMIs that the account owns as well as the AMIs that are shared with the account, so we have to have multiple calls

**Pull request merged**